content/guides/reader_conditionals.adoc

```diff
@@ -0,0 +1,198 @@
+
+== Host interop
+
+Host interop is one of the biggest pain points that cljx and reader conditionals both solve. You may have a Clojure file that is almost pure Clojure, but needs to call out to the host environment for one function. https://github.com/lymingtonprecision/route-ccrs/blob/c579aea05504736f2cfbd31c3c755f7e25fdad77/src/route_ccrs/manufacturing_methods.cljc#L8-L10[This] is a classic example:
+
+[source,clojure]
+----
+(defn str->int [s]
+  #?(:clj  (java.lang.Integer/parseInt s)
+     :cljs (js/parseInt s)))
+----
+
+== Namespaces
+
+Namespaces are the other big pain point for sharing code between Clojure and ClojureScript. ClojureScript has different syntax for https://github.com/clojure/clojurescript/wiki/Differences-from-Clojure#lisp[requiring macros] than Clojure. To use these macros in a `.cljc` file, you'll need Reader Conditionals in the namespace declaration.
+
+https://github.com/VesaKarvonen[Vesa Karvonen] has a https://github.com/VesaKarvonen/poc.cml/blob/6c80c55aa16ab1253ad3e85705b5360fbed73f7e/src/poc/cml/sem.cljc[neat trick] for this, taking advantage of the fact you can have multiple `:require`'s in one Clojure ns form. **N.B.: ClojureScript ns forms can only have one `:require` in them.**
+
+[source,clojure]
+----
+(ns poc.cml.sem
+  (#?(:clj :require :cljs :require-macros)
+    [poc.cml.macros :refer [go sync!]])
+  (:require
+    [poc.cml :refer [choose wrap]]
+    [poc.cml.util :refer [chan on put!]]))
+----
+
+Here is a larger example showing more conventional usage from a https://github.com/lymingtonprecision/route-ccrs/blob/c579aea05504736f2cfbd31c3c755f7e25fdad77/test/route_ccrs/schema/ids/part_no_test.cljc[test] in https://github.com/lymingtonprecision/route-ccrs[route-ccrs]
+
+[source,clojure]
+----
+(ns route-ccrs.schema.ids.part-no-test
+  (:require #?(:clj  [clojure.test :refer :all]
+               :cljs [cljs.test :refer-macros [is]])
+            #?(:cljs [cljs.test.check :refer [quick-check]])
+            #?(:clj  [clojure.test.check.clojure-test :refer [defspec]]
+               :cljs [cljs.test.check.cljs-test :refer-macros [defspec]])
+            #?(:clj  [clojure.test.check.properties :as prop]
+               :cljs [cljs.test.check.properties :as prop
+                       :include-macros true])
+            [schema.core :as schema :refer [check]]
+            [route-ccrs.schema.ids :refer [PartNo]]
+            [route-ccrs.generators.part-no
+             :refer [gen-part-no gen-invalid-part-no]]))
+----
+
+In this example, we want to be able to use the `rethinkdb.query` namespace in Clojure and ClojureScript. However we can't load the required `rethinkdb.net` in ClojureScript as it uses Java sockets to communicate with the database. Instead we use a reader conditional so it's only required when read by Clojure programs.
```

Comment:
- It wasn't clear to me on first read that this introduce another (third) example.

- s/so it's only required/so the the namespace is only required/